### ISSUE TITLE
Spaces inserted in Python placeholder 

### ISSUE THREAD

**Issue by sbrugman** [sqlfluff/sqlfluff#5219]
Hi there 👋 

Thanks for all the work on this awesome tool.
Running into an issue when linting t-sql with Python placeholders. 
The seemingly random spacing makes me think it's a bug, but it might just as well being a configuration mistake on my side.

### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running on a file with the following contents:
```sql
SELECT * from hello.{table};
```

### Expected Behaviour

Output of the file is formatted as:
```sql
SELECT * FROM hello.{table};
```

### Observed Behaviour

Output of the file is formatted as:
```sql
SELECT * FROM hello.{t ab le};
```

Spaces are inserted inside the placeholder.
The expected behaviour is obtained when I ignore the `jinja.padding` rule (however in this case explicitly the templater is not jinja). Is this intentional?

### How to reproduce

```system
sqlfluff fix --force --show-lint-violations --processes 0 --dialect tsql --templater python
```

### Dialect

tsql

### Version

sqlfluff, version 2.3.2
Python 3.8.16

### Configuration

```toml

[tool.sqlfluff.core]
max_line_length = 200
exclude_rules = ["ambiguous.column_count"]

[tool.sqlfluff.templater.python.context]
table = "my_table"

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"

[tool.sqlfluff.rules.capitalisation.functions]
extended_capitalisation_policy = "upper"

[tool.sqlfluff.rules.capitalisation.types]
capitalisation_policy = "upper"

[tool.sqlfluff.rules.capitalisation.identifiers]
extended_capitalisation_policy = "lower"
unquoted_identifiers_policy = "all"

[tool.sqlfluff.rules.convention.casting_style]
preferred_type_casting_style = "cast"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by josephwibowo** [comment]
having the same issue

**Comment by alanmcruickshank** [comment]
Yeah this is a bug, [JJ01](https://docs.sqlfluff.com/en/stable/rules.html#rule-jinja.padding) should be correcting it to:

```sql
SELECT * FROM hello.{ table };
```

**Comment by alanmcruickshank** [comment]
Oh hold up - I can see the problem here! A *jinja* padding rule is being applied to the *python* templater. I think we have two options here...

Either we disable this rule if the templater isn't `jinja` or a dependent of that. Alternatively we can make it work here. The python convention is usually _not_ to pad template strings, so I'm tempted to just disable this rule for the `python` templater.

@sbrugman @josephwibowo - do you have views? The workaround in the meantime is to exclude rule `JJ01`.
